src/worker.jl

```diff
@@ -854,7 +854,7 @@ function ensure_communicating(worker::Worker)
                         pop!(to_gather, dep2)
                     end
                 end
-                @sync gather_dep(worker, worker_addr, dep, to_gather, cause=key)
+                gather_dep(worker, worker_addr, dep, to_gather, cause=key)
```

Comment:
It was to fix a previous bug (in code that was deleted) that no longer appears so calling @sync isn't necessary anymore.